### PR TITLE
[WIP][Bugfix] Fix Qwen3-VL-Reranker wrong relevance scores due to incorrect pooling and attention type

### PR DESCRIPTION
**Pull request by haosdent**
## Purpose

Fix #35412

`SequenceClassificationMixin` (only used in the Transformers backend) hardcoded `default_seq_pooling_type = "CLS"`, which caused two issues for decoder-only models like Qwen3-VL-Reranker-2B:

1. **Wrong pooling type**: CLS extracts the first token, but decoder-only rerankers use the last token for scoring (`logits[:, -1, :]`).
2. **Wrong attention type**: Since `attn_type` derives from `default_seq_pooling_type`, CLS caused `"encoder_only"` (bidirectional) attention instead of `"decoder"` (causal) attention.

All encoder-based seq-cls models (BERT, RoBERTa, etc.) have native vLLM implementations and don't go through the Transformers backend. Only decoder-only models fall through to `SequenceClassificationMixin`, so the default should be `"LAST"`.

## Test Result

### CODE REVIEW

File: vllm/config/model.py [hunk 1]
```diff
@@ -1561,7 +1561,7 @@ def get_and_verify_max_len(self, max_model_len: int):
     def attn_type(self) -> AttnTypeStr:
         """Determine the attention type based on model configuration."""
         if self.pooler_config is not None:
-            seq_pooling_type = self._model_info.default_seq_pooling_type
+            seq_pooling_type = self.pooler_config.seq_pooling_type
             if seq_pooling_type == "CLS":
                 return "encoder_only"
```
Comment (noooop) [inline vllm/config/model.py:1566]: The logic here should be correct, please do not modify this part.

I believe the default here is "LAST".

Comment (noooop) [reply vllm/config/model.py:1566]: https://github.com/vllm-project/vllm/blob/07bdabef03c78c9dc6beb549185930888ec6f2ce/vllm/model_executor/models/interfaces_base.py#L214-L232

Comment (noooop) [reply vllm/config/model.py:1566]: https://github.com/vllm-project/vllm/blob/07bdabef03c78c9dc6beb549185930888ec6f2ce/vllm/model_executor/models/transformers/pooling.py#L48-L50

SequenceClassificationMixin is only used in the transformers backend

Comment (haosdent) [reply vllm/config/model.py:1566]: Thanks @noooop , have fixed

File: vllm/model_executor/models/transformers/pooling.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def __init__(self, *, vllm_config: "VllmConfig", prefix: str = ""):
 
 
 class SequenceClassificationMixin(SupportsCrossEncoding, VllmModelForPooling):
-    default_seq_pooling_type = "CLS"
+    default_seq_pooling_type = "LAST"
```
Comment (noooop) [inline vllm/model_executor/models/transformers/pooling.py:50]: ?

Please do not modify this, as it will affect all SequenceClassification (e.g. bert).

Is Qwen3-VL-Reranker's wrong relevance scores really due to incorrect pooling and attention type?

Comment (haosdent) [reply vllm/model_executor/models/transformers/pooling.py:50]: > Is Qwen3-VL-Reranker's wrong relevance scores really due to incorrect pooling and attention type?

OK, I think I misunderstood your point, let me continue to check the code to figure out what's wrong

Comment (haosdent) [reply vllm/model_executor/models/transformers/pooling.py:50]: Turns out it is due to parameter issues in user side. Sorry for the confusion , let me close the PR @noooop
